pkg/workload/workload.go

```diff
@@ -649,6 +649,11 @@ func ApplyAdmissionStatusPatch(ctx context.Context, c client.Client, patch *kueu
 	return c.Status().Patch(ctx, patch, client.Apply, client.FieldOwner(constants.AdmissionName))
 }
 
+// ApplyStatusPatch applies the patch of status fields of a workload with SSA.
+func ApplyStatusPatch(ctx context.Context, c client.Client, patch *kueue.Workload) error {
+	return c.Status().Patch(ctx, patch, client.Apply, client.FieldOwner(constants.WorkloadControllerName))
```

Comment:
I'm wondering if we need another entity to make the patch. What is the entity managing status Workload fields like `status.requeueState`, or changing the readiness of checks when transitioning from Pending -> Ready or Rejected?

Reply:
Regarding `status.requeueState` the entity is `kueue-admission`, and for transitions e.g. Pending -> Ready it's `kueue.x-k8s.io/provisioning-request`. Hence, I decided to go with `kueue-workload-controller` since this is the controller that calls this function. We can come up with some more consitent semantics for the field owners, but I believe this is out of the scope of this PR. WDYT?

Reply:
I see, this is fine from that perspective, but maybe we don't need to think about it if we update the checks along with setting Evicted=true, as in the other comment: https://github.com/kubernetes-sigs/kueue/pull/3323/files#r1818884218?

Reply:
Done